server/api/auth/setPW.post.ts

```diff
@@ -18,10 +21,20 @@ export default defineEventHandler(async (event) => {
       statusMessage: "Unknown User.",
     });
   }
-  const hashPW = await bcrypt.hash(body.password, 10);
+
+  if (user.salt === null) {
+    user.salt = generateSalt();
+  }
+
+  const hashPW = hashPassword(body.password, user.salt, SECRET_KEY);
 
   user.password = hashPW;
-  await user.save();
+  user.passwordChangedAt = DateTime.now().toISO();
 
-  return { sucesss: true };
+  await user.save();
```

Comment:
There's no error handling in the case that the transaction failed.  It's fine that we don't cause the error gets bubbled up, but we should at least make sure that the UI shows that its a 500 failure instead of gaslighting the user into thinking that it's an issue on their end or not providing any info at all (the error modal pops up with no message inside).